frontend/apps/app/components/SessionDetailPage/components/Output/components/Artifact/hooks/useRealtimeArtifact.ts

```diff
@@ -14,7 +14,10 @@ const artifactDataSchema = v.object({
   updated_at: v.string(),
 })
 
-export function useRealtimeArtifact(designSessionId: string) {
+export function useRealtimeArtifact(
+  designSessionId: string,
+  onChangeArtifact?: (artifact: Artifact | null) => void,
```

Comment:
Is this not acceptable?

```suggestion
  onChangeArtifact: (artifact: Artifact) => void,
```